src/RoomDirectoryModel.h

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <QAbstractListModel>
+#include <QHash>
+#include <QString>
+#include <string>
+#include <vector>
+
+#include "MatrixClient.h"
+#include <mtx/responses/public_rooms.hpp>
+#include <mtxclient/http/errors.hpp>
+
+#include "Logging.h"
+
+namespace mtx::http {
+using RequestErr = const std::optional<mtx::http::ClientError> &;
+}
+namespace mtx::responses {
+struct PublicRooms;
+}
+
+class RoomDirectoryModel : public QAbstractListModel
+{
+        Q_OBJECT
+
+public:
+        explicit RoomDirectoryModel(QObject *parent = nullptr, const std::string &s = "");
+
+        enum Roles
+        {
+                Name = Qt::UserRole,
+                Id,
+                AvatarUrl,
+                Topic,
+                MemberCount,
+                Previewable
+        };
+        virtual QHash<int, QByteArray> roleNames() const override;
+
+        virtual QVariant data(const QModelIndex &index, int role) const override;
+
+        virtual inline int rowCount(
+          [[maybe_unused]] const QModelIndex &parent = QModelIndex()) const override
+        {
+                return static_cast<int>(publicRoomsData_.size());
+        }
+
+        virtual inline bool canFetchMore([[maybe_unused]] const QModelIndex &parent) const override
+        {
+                nhlog::net()->debug("determining if can fetch more");
+                return canFetchMore_;
+        }
+        virtual void fetchMore([[maybe_unused]] const QModelIndex &parent) override;
```

Comment:
No need for virtual, if you already have override. Also, you can remove the parameter names for the cases, where you don't default init the param. That way you don't need the maybe_unused. (i.e. `([[maybe_unused]] const QModelIndex &parent)` -> `(const QModelIndex &)`)